worker/docker_test.go

```diff
@@ -1,12 +1,146 @@
 package worker
 
 import (
+	"context"
+	"errors"
 	"fmt"
+	"io"
+	"strings"
 	"testing"
 
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/api/types/network"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
 )
 
+type MockDockerClient struct {
+	mock.Mock
+}
+
+func (m *MockDockerClient) ImagePull(ctx context.Context, ref string, options image.PullOptions) (io.ReadCloser, error) {
```

Comment:
FTR to reduce the boilerplate here, we can also only add to the interface the exact functions that we need from the client. Then we can mock them on a need-by-need basiss

Reply:
Hi @victorges,

I’ve ensured that all tests related to [[this branch](https://github.com/livepeer/ai-worker/tree/auto_pull_containers)](https://github.com/livepeer/ai-worker/tree/auto_pull_containers) have been implemented based on your changes. Please take a look, and feel free to make any adjustments if you see room for improvement.

I also attempted to remove the [[ContainerInspect](https://github.com/livepeer/ai-worker/blob/72bca183d51b4437914ec31e994039298a0204d2/worker/docker_test.go#L46)](https://github.com/livepeer/ai-worker/blob/72bca183d51b4437914ec31e994039298a0204d2/worker/docker_test.go#L46) mock method since it’s not being used in the tests. However, doing so caused the following error:

```
cannot use mockDockerClient (variable of type *MockDockerClient) as DockerClient value in struct literal: *MockDockerClient does not implement DockerClient (missing method ContainerInspect)
```

Let me know if you have any suggestions on how to address this, or if there’s a better way to handle it.

Thanks!

Reply:
Really curious that it would complain after removing the function from the interface. Let's leave it anyway since I'm using it in some of the latest changes I made to ai-worker for realtime containers.

Reply:
If you already use it in the Manager for realtime let's keep it as is.